Jenkinsfile

```diff
@@ -32,13 +32,22 @@ node('rhel7'){
         packageJson.extensionDependencies = ["ms-kubernetes-tools.vscode-kubernetes-tools"]
         writeJSON file: 'package.json', json: packageJson, pretty: 4
         sh "vsce package -o openshift-connector-${packageJson.version}-${env.BUILD_NUMBER}.vsix"
+		def findVsix = findFiles(glob: '**.vsix')
+		sh "sha256sum ${findVsix[0].path} > openshift-connector-${packageJson.version}-${env.BUILD_NUMBER}.vsix.sha256"
 	}
 
 	if(params.UPLOAD_LOCATION) {
 		stage('Snapshot') {
-			def filesToPush = findFiles(glob: '**.vsix')
-			sh "rsync -Pzrlt --rsh=ssh --protocol=28 ${filesToPush[0].path} ${UPLOAD_LOCATION}/snapshots/vscode-openshift-tools/"
-            stash name:'vsix', includes:filesToPush[0].path
+			def filesToPush = findFiles(glob: '**.vsix*')
+			for (int i = 0; i < filesToPush.size(); i++) {
+				sh "rsync -Pzrlt --rsh=ssh --protocol=28 ${filesToPush[i].path} ${UPLOAD_LOCATION}/stable/vscode-openshift-tools/"
```

Comment:
probably it can be done even simpler. Just use 

`sh "rsync -Pzrlt --rsh=ssh --protocol=28 *.vsix* ${UPLOAD_LOCATION}/stable/vscode-openshift-tools/"` 

if `*.vsix*` does not work properly just use two sh calls

`sh "rsync -Pzrlt --rsh=ssh --protocol=28 *.vsix ${UPLOAD_LOCATION}/stable/vscode-openshift-tools/"` 
`sh "rsync -Pzrlt --rsh=ssh --protocol=28 *.sha256 ${UPLOAD_LOCATION}/stable/vscode-openshift-tools/"` 

assuming the default folder is root of git repository.

If all above works you don't need to search for files, stash, iterate over search results. It would be just sh call or calls.